inference-engine/tests/functional/plugin/myriad/subgraph_tests/dsr_matmul.cpp

```diff
@@ -59,78 +60,50 @@ using Parameters = std::tuple<
 >;
 
 class DSR_MatMul : public testing::WithParamInterface<Parameters>,
-        public LayerTestsUtils::LayerTestsCommon {
+                   public DSR_TestsCommon {
 protected:
-    ngraph::ParameterVector setting_up_input_dynamism(
-            const DYNAMISM_MODE mode,
-            const std::shared_ptr<ngraph::opset3::Parameter> input_A,
-            const std::shared_ptr<ngraph::opset3::Parameter> input_B,
-            std::shared_ptr<ngraph::Node>& renewed_input_A,
-            std::shared_ptr<ngraph::Node>& renewed_input_B,
-            std::shared_ptr<ngraph::Node>& A_shape_node,
-            std::shared_ptr<ngraph::Node>& B_shape_node) const {
-        ngraph::ParameterVector parameters{input_A, input_B};
-
-        auto input_A_dsr = std::make_shared<ngraph::opset3::Parameter>(ngraph::element::i64, ngraph::Shape{input_A->get_shape().size()});
-        auto input_B_dsr = std::make_shared<ngraph::opset3::Parameter>(ngraph::element::i64, ngraph::Shape{input_B->get_shape().size()});
-
-        auto dsr_A = std::make_shared<ngraph::vpu::op::DynamicShapeResolver>(input_A, input_A_dsr);
-        auto dsr_B = std::make_shared<ngraph::vpu::op::DynamicShapeResolver>(input_B, input_B_dsr);
+    std::shared_ptr<ngraph::Node> createTestedOp() override {
+        const auto& params = GetParam();
+        const auto& mode = std::get<0>(params);
+        const auto& data_type = std::get<1>(params);
+        const auto& matmul_setup = std::get<2>(params);
+        targetDevice = std::get<3>(params);
+
+        std::shared_ptr<ngraph::Node> inputA, inputB;
 
         switch (mode) {
             case DYNAMISM_MODE::BOTH_INPUTS_DYNAMIC: {
-                parameters.push_back(input_A_dsr);
-                parameters.push_back(input_B_dsr);
-                renewed_input_A = dsr_A;
-                renewed_input_B = dsr_B;
-                A_shape_node = input_A_dsr;
-                B_shape_node = input_B_dsr;
+                inputA = createInputSubgraphWithDSR(data_type, DataShapeWithUpperBound{matmul_setup.A.realShape, matmul_setup.A.upperBoundShape});
+                inputB = createInputSubgraphWithDSR(data_type, DataShapeWithUpperBound{matmul_setup.B.realShape, matmul_setup.B.upperBoundShape});
                 break;
             }
             case DYNAMISM_MODE::A_INPUT_DYNAMIC: {
-                parameters.push_back(input_A_dsr);
-                renewed_input_A = dsr_A;
-                renewed_input_B = input_B;
-                A_shape_node = input_A_dsr;
-                B_shape_node = ngraph::opset3::Constant::create(ngraph::element::i64, {input_B->get_shape().size()}, input_B->get_shape());
+                inputA = createInputSubgraphWithDSR(data_type, DataShapeWithUpperBound{matmul_setup.A.realShape, matmul_setup.A.upperBoundShape});
+                const auto inputBParam = std::make_shared<ngraph::opset3::Parameter>(data_type, matmul_setup.B.realShape);
```

Comment:
Why we need `inputBParam`? Can't we explicitly assign it to inputB?

Reply:
The problem there was in necessity of inserting inputB into `m_parameterVector` vector for further Function creation. While inputB is `std::shared_ptr<ngraph::Node>`, `ParameterVector` accepts only `Parameter`s. I've refactored this and added an appropriate method into `DSR_TestsCommon` base class. Please take a look.